src/Raven.Server/Documents/Replication/Incoming/IncomingReplicationHandler.cs

```diff
@@ -1361,7 +685,8 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
             protected virtual void SaveSourceEtag(DocumentsOperationContext context)
             {
                 context.LastReplicationEtagFrom ??= new Dictionary<string, long>();
-                context.LastReplicationEtagFrom[_replicationInfo.SourceDatabaseId] = _lastEtag;
+                if (_replicationType != ReplicationLatestEtagRequest.ReplicationType.Sharded)
```

Comment:
remove this?

Reply:
done (I added a check for `_replicationInfo.SourceDatabaseId != null `because we might get null for Sharded)